commit 1ca9fda4b2f3413e4bc748b983f5585629ca0560
Author: Chris McDermott <lcm@linux.vnet.ibm.com>
Date:   Thu Jul 24 19:06:09 2008 -0700

    x86: fix IBM Summit based systems' phys_cpu_present_map on 32-bit kernels
    
    x86 kernels on IBM Summit based systems will only online 1 CPU because the
    phys_cpu_present_map is not set up correctly. Patch below applied to
    2.6.26-git10.
    
    Signed-off-by: Chris McDermott <lcm@linux.vnet.ibm.com>
    Tested-by: Tim Pepper <lnxninga@linux.vnet.ibm.com>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

diff --git a/include/asm-x86/mach-summit/mach_apic.h b/include/asm-x86/mach-summit/mach_apic.h
index 75d2c95..c47e2ab 100644
--- a/include/asm-x86/mach-summit/mach_apic.h
+++ b/include/asm-x86/mach-summit/mach_apic.h
@@ -122,7 +122,7 @@ static inline physid_mask_t ioapic_phys_id_map(physid_mask_t phys_id_map)
 
 static inline physid_mask_t apicid_to_cpu_present(int apicid)
 {
-	return physid_mask_of_physid(0);
+	return physid_mask_of_physid(apicid);
 }
 
 static inline void setup_portio_remap(void)
